src/plugins/orangehrmCoreOAuthPlugin/Repository/ClientRepository.php

```diff
@@ -46,6 +57,30 @@ public function getClientEntity($clientIdentifier): ?ClientEntityInterface
      */
     public function validateClient($clientIdentifier, $clientSecret, $grantType): bool
     {
-        throw new Exception(__METHOD__);
+        $oauthClient = $this->getOAuthClientById($clientIdentifier);
+        if (!$oauthClient instanceof OAuthClient) {
+            return false;
+        }
+        if ($oauthClient->isConfidential() === true && $oauthClient->getClientSecret() === null) {
+            // Confidential client must have a secret
+            return false;
+        }
+
+        if ($grantType === 'refresh_token' && $oauthClient->isConfidential() === false) {
+            return true;
+        }
+
+        return $this->validateClientSecret($oauthClient, new UserCredential(null, $clientSecret));
+    }
+
+    /**
+     * @param OAuthClient $client
+     * @param UserCredentialInterface $givenCred
+     * @return bool
+     */
+    private function validateClientSecret(OAuthClient $client, UserCredentialInterface $givenCred): bool
+    {
+        // TODO:: handle encryption or hashing
```

Comment:
TODO:: need to check whether we need to handle confidential clients as we are going to support only auth code grant type

https://www.oauth.com/oauth2-servers/client-registration/client-id-secret